From 0983f081efada98a6b61d70de10bbf01d5f430bd Mon Sep 17 00:00:00 2001
From: Jami Kettunen <jamipkettunen@gmail.com>
Date: Sun, 16 Oct 2022 20:51:39 +0300
Subject: [PATCH] makefile: add installmisc target to install everything except
 binaries

The "install" target still behaves like before but now one can
"make installmisc" to not install the binaries in case they're being
handled (built and installed) outside the makefile in e.g. a
distribution's package build environment.

Link: https://lists.sr.ht/~craftyguy/superd/patches/36158
---
 Makefile | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 0a7b0d2..8bd6946 100644
--- a/Makefile
+++ b/Makefile
@@ -58,12 +58,14 @@ doc: $(DOCS)
 clean:
 	$(RM) $(DOCS) superd superctl
 
-install: $(DOCS) superd superctl
+installbins: superd superctl
+	install -Dm755 superd -t $(DESTDIR)$(BINDIR)/
+	install -Dm755 superctl -t $(DESTDIR)$(BINDIR)/
+
+installmisc: $(DOCS)
 	mkdir -m755 -p \
 		$(DESTDIR)$(SYSCONFDIR)/superd/services \
 		$(DESTDIR)$(SHAREDIR)/superd/services
-	install -Dm755 superd -t $(DESTDIR)$(BINDIR)/
-	install -Dm755 superctl -t $(DESTDIR)$(BINDIR)/
 	install -Dm644 superd.1 -t $(DESTDIR)$(MANDIR)/man1/
 	install -Dm644 superd.service.5 -t $(DESTDIR)$(MANDIR)/man5/
 	install -Dm644 superctl.1 -t $(DESTDIR)$(MANDIR)/man1/
@@ -72,6 +74,8 @@ install: $(DOCS) superd superctl
 	install -Dm644 completions/bash/superctl \
 		$(DESTDIR)$(SHAREDIR)/bash-completion/completions/superctl
 
+install: installbins installmisc
+
 .PHONY: checkinstall
 checkinstall:
 	$(DESTDIR)$(BINDIR)/superd -v
-- 
2.38.0

